lib/Internal/Posix/Handle.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Amp\Process\Internal\Posix;
+
+use Amp\Deferred;
+use Amp\Process\Internal\ProcessHandle;
+
+final class Handle extends ProcessHandle {
+    public function __construct() {
+        $this->pidDeferred = new Deferred;
+        $this->joinDeferred = new Deferred;
+        $this->originalParentPid = \getmypid();
+    }
+
+    /** @var Deferred */
+    public $joinDeferred;
```

Comment:
this really is an `exitCodeDeferred`, right? `joinDeferred` does not tell me much about what will be returned from it.

Reply:
The method returning the exit code is called `join` as well.

Reply:
I am aware, I just found it confusing while reading the code.